pkg/ottl/expression_test.go

```diff
@@ -1550,3 +1552,85 @@ func Test_StandardPMapGetter_WrappedError(t *testing.T) {
 	_, ok := err.(TypeError)
 	assert.False(t, ok)
 }
+
+func Test_StandardDurationGetter(t *testing.T) {
+	oneHourOneMinuteOneSecond, err := time.ParseDuration("1h1m1s")
+	require.NoError(t, err)
+
+	oneHundredNsecs, err := time.ParseDuration("100ns")
+	require.NoError(t, err)
+
+	tenMilliseconds, err := time.ParseDuration("10ms66us7000ns")
+	require.NoError(t, err)
+
+	tests := []struct {
+		name             string
+		getter           StandardDurationGetter[interface{}]
+		want             interface{}
+		valid            bool
+		expectedErrorMsg string
+	}{
+		{
+			name: "complex duration",
+			getter: StandardDurationGetter[interface{}]{
+				Getter: func(ctx context.Context, tCtx interface{}) (interface{}, error) {
+					return time.ParseDuration("1h1m1s")
+				},
+			},
+			want:  oneHourOneMinuteOneSecond,
+			valid: true,
+		},
+		{
+			name: "simple duration",
+			getter: StandardDurationGetter[interface{}]{
+				Getter: func(ctx context.Context, tCtx interface{}) (interface{}, error) {
+					return time.ParseDuration("100ns")
+				},
+			},
+			want:  oneHundredNsecs,
+			valid: true,
+		},
+		{
+			name: "complex duation values less than 1 seconc",
+			getter: StandardDurationGetter[interface{}]{
+				Getter: func(ctx context.Context, tCtx interface{}) (interface{}, error) {
+					return time.ParseDuration("10ms66us7000ns")
+				},
+			},
+			want:  tenMilliseconds,
+			valid: true,
+		},
+		{
+			name: "invalid duration units",
+			getter: StandardDurationGetter[interface{}]{
+				Getter: func(ctx context.Context, tCtx interface{}) (interface{}, error) {
+					return time.ParseDuration("70ps")
+				},
+			},
+			valid:            false,
+			expectedErrorMsg: "unknown unit",
+		},
+		{
+			name: "wrong type - int",
```

Comment:
Lets add one more test scenario for `nil`.

Reply:
Added!